pkg/image/api/helper.go

```diff
@@ -307,6 +307,38 @@ func AddTagEventToImageStream(stream *ImageStream, tag string, next TagEvent) bo
 	return true
 }
 
+// UpdateChangedTrackingTags identifies any tags in the status that have changed and
+// ensures any referenced tracking tags are also updated. It returns the number of
+// updates applied.
+func UpdateChangedTrackingTags(new, old *ImageStream) int {
+	changes := 0
+	for newTag, newImages := range new.Status.Tags {
+		if oldImages, ok := old.Status.Tags[newTag]; ok {
```

Comment:
When?  From tagsChanged?

On Oct 21, 2015, at 3:55 AM, Michal Minar <notifications@github.com> wrote:

In pkg/image/api/helper.go
<https://github.com/openshift/origin/pull/5267#discussion_r42592917>:

> @@ -307,6 +307,38 @@ func AddTagEventToImageStream(stream *ImageStream, tag string, next TagEvent) bo
>  	return true
>  }
>
> +// UpdateChangedTrackingTags identifies any tags in the status that have changed and
> +// ensures any referenced tracking tags are also updated. It returns the number of
> +// updates applied.
> +func UpdateChangedTrackingTags(new, old *ImageStream) int {
> +	changes := 0
> +	for newTag, newImages := range new.Status.Tags {
> +		if oldImages, ok := old.Status.Tags[newTag]; ok {

old may be nil.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5267/files#r42592917>.